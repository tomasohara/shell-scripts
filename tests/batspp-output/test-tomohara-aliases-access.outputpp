#!/usr/bin/env bats


# Make executables ./tests/../ visible to PATH
PATH="/home/aveey/tom-project/shell-scripts/tests/batspp-only/../:$PATH"

# Enable aliases
shopt -s expand_aliases



@test "test-1" {
	testfolder="/tmp/batspp-72160/test-1"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'export TEMP_TOM_BIN=tmp/tomasohara-shell-scripts\n' "=========="
	test-1-actual 
	echo "=========" $'$ export TOM_ALIAS_FILE="$TEMP_TOM_BIN/tomohara-aliases.bash"\n$ mkdir -p "$TEMP_TOM_BIN"' "========="
	test-1-expected 
	echo "============================"
	# ???: 'export TEMP_TOM_BIN=tmp/tomasohara-shell-scripts\n'=$(test-1-actual)
	# ???: '$ export TOM_ALIAS_FILE="$TEMP_TOM_BIN/tomohara-aliases.bash"\n$ mkdir -p "$TEMP_TOM_BIN"'=$(test-1-expected)
	[ "$(test-1-actual)" == "$(test-1-expected)" ]
}

function test-1-actual () {
	# no-op in case content just a comment
	true

	export TEMP_TOM_BIN=tmp/tomasohara-shell-scripts

}

function test-1-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
$ export TOM_ALIAS_FILE="$TEMP_TOM_BIN/tomohara-aliases.bash"
$ mkdir -p "$TEMP_TOM_BIN"
END_EXPECTED
}


@test "test-3" {
	testfolder="/tmp/batspp-72160/test-3"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'curl https://raw.githubusercontent.com/tomasohara/shell-scripts/main/tomohara-aliases.bash > "$TOM_ALIAS_FILE"\n' "=========="
	test-3-actual 
	echo "=========" $'% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n100  137k  100  137k    0     0   194k      0 --:--:-- --:--:-- --:--:--  194k' "========="
	test-3-expected 
	echo "============================"
	# ???: 'curl https://raw.githubusercontent.com/tomasohara/shell-scripts/main/tomohara-aliases.bash > "$TOM_ALIAS_FILE"\n'=$(test-3-actual)
	# ???: '% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n100  137k  100  137k    0     0   194k      0 --:--:-- --:--:-- --:--:--  194k'=$(test-3-expected)
	[ "$(test-3-actual)" == "$(test-3-expected)" ]
}

function test-3-actual () {
	# no-op in case content just a comment
	true

	curl https://raw.githubusercontent.com/tomasohara/shell-scripts/main/tomohara-aliases.bash > "$TOM_ALIAS_FILE"

}

function test-3-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  137k  100  137k    0     0   194k      0 --:--:-- --:--:-- --:--:--  194k
END_EXPECTED
}


@test "test-4" {
	testfolder="/tmp/batspp-72160/test-4"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'file "$TOM_ALIAS_FILE"\n' "=========="
	test-4-actual 
	echo "=========" $'tmp/tomasohara-shell-scripts/tomohara-aliases.bash: Bourne-Again shell script, UTF-8 Unicode text executable, with very long lines' "========="
	test-4-expected 
	echo "============================"
	# ???: 'file "$TOM_ALIAS_FILE"\n'=$(test-4-actual)
	# ???: 'tmp/tomasohara-shell-scripts/tomohara-aliases.bash: Bourne-Again shell script, UTF-8 Unicode text executable, with very long lines'=$(test-4-expected)
	[ "$(test-4-actual)" == "$(test-4-expected)" ]
}

function test-4-actual () {
	# no-op in case content just a comment
	true

	file "$TOM_ALIAS_FILE"

}

function test-4-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
tmp/tomasohara-shell-scripts/tomohara-aliases.bash: Bourne-Again shell script, UTF-8 Unicode text executable, with very long lines
END_EXPECTED
}


@test "test-5" {
	testfolder="/tmp/batspp-72160/test-5"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'export BATCH_MODE=1\n' "=========="
	test-5-actual 
	echo "=========" $'$ source "$TOM_ALIAS_FILE"' "========="
	test-5-expected 
	echo "============================"
	# ???: 'export BATCH_MODE=1\n'=$(test-5-actual)
	# ???: '$ source "$TOM_ALIAS_FILE"'=$(test-5-expected)
	[ "$(test-5-actual)" == "$(test-5-expected)" ]
}

function test-5-actual () {
	# no-op in case content just a comment
	true

	export BATCH_MODE=1

}

function test-5-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
$ source "$TOM_ALIAS_FILE"
END_EXPECTED
}


@test "test-6" {
	testfolder="/tmp/batspp-72160/test-6"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'show-macros | wc -l\n' "=========="
	test-6-actual 
	echo "=========" $'4161' "========="
	test-6-expected 
	echo "============================"
	# ???: 'show-macros | wc -l\n'=$(test-6-actual)
	# ???: '4161'=$(test-6-expected)
	[ "$(test-6-actual)" == "$(test-6-expected)" ]
}

function test-6-actual () {
	# no-op in case content just a comment
	true

	show-macros | wc -l

}

function test-6-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
4161
END_EXPECTED
}


@test "test-7" {
	testfolder="/tmp/batspp-72160/test-7"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'rm -rf "$TEMP_TOM_BIN"\n' "=========="
	test-7-actual 
	echo "=========" $"removed 'tmp/tomasohara-shell-scripts/tomohara-aliases.bash'\nremoved directory 'tmp/tomasohara-shell-scripts'" "========="
	test-7-expected 
	echo "============================"
	# ???: 'rm -rf "$TEMP_TOM_BIN"\n'=$(test-7-actual)
	# ???: "removed 'tmp/tomasohara-shell-scripts/tomohara-aliases.bash'\nremoved directory 'tmp/tomasohara-shell-scripts'"=$(test-7-expected)
	[ "$(test-7-actual)" == "$(test-7-expected)" ]
}

function test-7-actual () {
	# no-op in case content just a comment
	true

	rm -rf "$TEMP_TOM_BIN"

}

function test-7-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
removed 'tmp/tomasohara-shell-scripts/tomohara-aliases.bash'
removed directory 'tmp/tomasohara-shell-scripts'
END_EXPECTED
}
