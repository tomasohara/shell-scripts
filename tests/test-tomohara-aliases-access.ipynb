{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Warning: this creates a temporary directory for aliases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Set directory for alias work\n",
    "export TEMP_TOM_BIN=tmp/tomasohara-shell-scripts\n",
    "export TOM_ALIAS_FILE=\"$TEMP_TOM_BIN/tomohara-aliases.bash\"\n",
    "mkdir -p \"$TEMP_TOM_BIN\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100  137k  100  137k    0     0   194k      0 --:--:-- --:--:-- --:--:--  194k\n"
     ]
    }
   ],
   "source": [
    "# Get latest version of aliases\n",
    "curl https://raw.githubusercontent.com/tomasohara/shell-scripts/main/tomohara-aliases.bash > \"$TOM_ALIAS_FILE\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tmp/tomasohara-shell-scripts/tomohara-aliases.bash: Bourne-Again shell script, UTF-8 Unicode text executable, with very long lines\n"
     ]
    }
   ],
   "source": [
    "file \"$TOM_ALIAS_FILE\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Note: batch mode needed for Jupyter usage\n",
    "export BATCH_MODE=1\n",
    "source \"$TOM_ALIAS_FILE\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4161\n"
     ]
    }
   ],
   "source": [
    "show-macros | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "removed 'tmp/tomasohara-shell-scripts/tomohara-aliases.bash'\n",
      "removed directory 'tmp/tomasohara-shell-scripts'\n"
     ]
    }
   ],
   "source": [
    "rm -rf \"$TEMP_TOM_BIN\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
