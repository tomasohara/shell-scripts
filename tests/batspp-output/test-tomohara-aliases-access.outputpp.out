1..6
not ok 1 test-1
# (in test file batspp-output/test-tomohara-aliases-access.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== export TEMP_TOM_BIN=tmp/tomasohara-shell-scripts
#  ==========
# ========= $ export TOM_ALIAS_FILE="$TEMP_TOM_BIN/tomohara-aliases.bash"
# $ mkdir -p "$TEMP_TOM_BIN" =========
# $ export TOM_ALIAS_FILE="tmp/tomasohara-shell-scripts/tomohara-aliases.bash"
# $ mkdir -p "tmp/tomasohara-shell-scripts"
# ============================
not ok 2 test-3
# (from function `test-3-actual' in file batspp-output/test-tomohara-aliases-access.outputpp, line 67,
#  in test file batspp-output/test-tomohara-aliases-access.outputpp, line 54)
#   `test-3-actual ' failed
# ========== curl https://raw.githubusercontent.com/tomasohara/shell-scripts/main/tomohara-aliases.bash > "$TOM_ALIAS_FILE"
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/test-tomohara-aliases-access.outputpp: line 67: : No such file or directory
not ok 3 test-4
# (in test file batspp-output/test-tomohara-aliases-access.outputpp, line 96)
#   `[ "$(test-4-actual)" == "$(test-4-expected)" ]' failed
# ========== file "$TOM_ALIAS_FILE"
#  ==========
# cannot open `' (No such file or directory)
# ========= tmp/tomasohara-shell-scripts/tomohara-aliases.bash: Bourne-Again shell script, UTF-8 Unicode text executable, with very long lines =========
# tmp/tomasohara-shell-scripts/tomohara-aliases.bash: Bourne-Again shell script, UTF-8 Unicode text executable, with very long lines
# ============================
not ok 4 test-5
# (in test file batspp-output/test-tomohara-aliases-access.outputpp, line 130)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== export BATCH_MODE=1
#  ==========
# ========= $ source "$TOM_ALIAS_FILE" =========
# $ source ""
# ============================
not ok 5 test-6
# (in test file batspp-output/test-tomohara-aliases-access.outputpp, line 164)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== show-macros | wc -l
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/test-tomohara-aliases-access.outputpp: line 171: show-macros: command not found
# 0
# ========= 4161 =========
# 4161
# ============================
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/test-tomohara-aliases-access.outputpp: line 171: show-macros: command not found
not ok 6 test-7
# (in test file batspp-output/test-tomohara-aliases-access.outputpp, line 198)
#   `[ "$(test-7-actual)" == "$(test-7-expected)" ]' failed
# ========== rm -rf "$TEMP_TOM_BIN"
#  ==========
# ========= removed 'tmp/tomasohara-shell-scripts/tomohara-aliases.bash'\nremoved directory 'tmp/tomasohara-shell-scripts' =========
# removed 'tmp/tomasohara-shell-scripts/tomohara-aliases.bash'
# removed directory 'tmp/tomasohara-shell-scripts'
# ============================
